vector/src/main/java/im/vector/app/features/permalink/PermalinkHandler.kt

```diff
@@ -89,7 +90,13 @@ class PermalinkHandler @Inject constructor(private val activeSessionHolder: Acti
 
                 val rootThreadEventId = permalinkData.eventId?.let { eventId ->
                     val room = roomId?.let { session?.getRoom(it) }
-                    room?.getTimelineEvent(eventId)?.root?.getRootThreadEventId()
+
+                    val rootThreadEventId = room?.getTimelineEvent(eventId)?.root?.getRootThreadEventId()
+                    rootThreadEventId ?: if (room?.getTimelineEvent(eventId)?.isRootThread() == true) {
+                        eventId
+                    } else {
+                        null
+                    }
```

Comment:
I understand the code and it should work, hence it's a bit confusing to have multiple declaration of `val rootThreadEventId`. Not a blocker anyway.